### PR TITLE
Enable EnsembleClassifier and EnsembleRegressor

### PR DESCRIPTION
**Pull request by najeeb-kazmi**
Partial #199 

Also added are components for feature and row subset selction to pass to each trainer in the ensemble, sub model selction for pruning the trained ensemble, and output combiners for aggregating the outputs of all the models in the ensemble.

### CODE REVIEW

File: src/python/nimbusml/examples/EnsembleClassifier.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# 1   42     1    0-5yrs        1       1 ...       2            0  ...
+# 2   39     1    0-5yrs        2       6 ...       3            0  ...
+# 3   34     1    0-5yrs        2       4 ...       4            0  ...
+# 4   35     1   6-11yrs        1       3 ...       5            1  ...
+
+
+# define the training pipeline
+pipeline = Pipeline([
+    OneHotVectorizer(columns={'edu': 'education'}),
+    EnsembleClassifier(feature=['age', 'edu', 'parity'],
```
Comment (ganik) [inline src/python/nimbusml/examples/EnsembleClassifier.py]: >EnsembleClassifier [](start = 4, length = 18)

What type of predictor is used by default?

Comment (najeeb-kazmi) [reply src/python/nimbusml/examples/EnsembleClassifier.py]: Multiclass Logistic Regression (LbfgsMaximumEntropyMulticlassTrainer) for multiclass, OnlineGradientDescentTrainer for regression. Right now, this is not configurable - BasePredictor parameter is not exposed in entrypoints from ML.NET side.

---
In reply to: [310263451](https://github.com/microsoft/NimbusML/pull/207#discussion_r310263451) [](ancestors = 310263451)

Comment (ganik) [reply src/python/nimbusml/examples/EnsembleClassifier.py]: could you add this ensemble documentation 


---
In reply to: [310280620](https://github.com/microsoft/NimbusML/pull/207#discussion_r310280620) [](ancestors = 310280620,310263451)

File: src/python/tools/entrypoint_compiler.py [hunk 2]
```diff
@@ -302,24 +302,13 @@ def write_api(entrypoint, kind="node", pkg_path=None, overwrite=False):
...
-    dots = "..."
-    if "." in class_dir:
-        dots = "...."
-    core_class_imports = [
-        arg.get_import(
-            prefix=(
-                    "%sentrypoints." %
-                    dots)) for arg in visible_args if
-        arg.get_import() is not None]
-    core_class_imports = '\n'.join(core_class_imports)
```
Comment (najeeb-kazmi) [inline src/python/tools/entrypoint_compiler.py]: Duplicate code

File: src/python/nimbusml/examples/EnsembleRegressor.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# define the training pipeline
+pipeline = Pipeline([
+    OneHotVectorizer(columns={'edu': 'education'}),
+    EnsembleRegressor(feature=['induced', 'edu'], label='age', num_models=3,
+                      sub_model_selector_type=RegressorBestPerformanceSelector(),
+                      output_combiner=RegressorMedian())
+])
+
+# train, predict, and evaluate
+# TODO: Replace with CV
```
Comment (ganik) [inline src/python/nimbusml/examples/EnsembleRegressor.py]: remove

File: src/python/nimbusml/examples/EnsembleClassifier.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                       label='induced',
+                       num_models=3,
+                       sub_model_selector_type=ClassifierBestPerformanceSelector(),
+                       output_combiner=ClassifierVoting())
+])
+
+# train, predict, and evaluate
+# TODO: Replace with CV
```
Comment (ganik) [inline src/python/nimbusml/examples/EnsembleClassifier.py]: remove

**Pull request merged**